[VectorCombine] Do not try to operate on OperandBundles.

davemgreen opened the pull request:
This bails out if we see an intrinsic with an operand bundle on it, to make sure we don't process the bundles incorrectly.

Fixes #110382.

davemgreen commented:
@Zentrik, @Heresh83 @RKSimon FYI.

RKSimon commented on llvm/lib/Transforms/Vectorize/VectorCombine.cpp:
```diff
@@ -2039,6 +2039,8 @@ bool VectorCombine::foldShuffleToIdentity(Instruction &I) {
         continue;
       } else if (auto *II = dyn_cast<IntrinsicInst>(FrontU);
                  II && isTriviallyVectorizable(II->getIntrinsicID())) {
+        if (II->hasOperandBundles())
```
can we not put this in the if condition above?

davemgreen replied:
I've updated to check both the first lane and the others, in case the intrinsic is different.

RKSimon reviewed (approved):
LGTM with one minor query

davemgreen committed:
[VectorCombine] Do not try to operate on OperandBundles.

This bails out if we see an intrinsic with an operand bundle on it, to make
sure we don't process the bundles incorrectly.

Fixes #110382.
